lang/funcs/core/datetime/format_func.go

```diff
@@ -0,0 +1,51 @@
+// Mgmt
+// Copyright (C) 2013-2019+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package coredatetime
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/purpleidea/mgmt/lang/funcs/simple"
+	"github.com/purpleidea/mgmt/lang/types"
+)
+
+func init() {
+	simple.ModuleRegister(ModuleName, "format", &types.FuncValue{
+		T: types.NewType("func(a int, b str) str"),
```

Comment:
Should this be str, int to match the style of printf?

Reply:
No, as it is different from printf. I would prefer to see the "date" first, then the format. Just like if we implement other functions with date, I would like to always have the date first.

Imagine we implement    ` func (t Time) After(u Time) bool`, I would like it to be:
`func(t int, u int) bool`

Reply:
Okay fine.